### DISCUSSION TITLE
Unable to transmit messages between two VMs (OpenDDS 23.1 Messenger example)

### DISCUSSION THREAD

**Discussion by kswain1906** [OpenDDS/OpenDDS#4263] [Q&A]
I am able to run the OpenDDS 23.1 Messenger example on a single RHEL 8.3 VM with DCPSInfoRepo configuration.  These are the steps I follow:
1) In one terminal window: source setenv.sh and start server with DCPSInfoRepo -ORBListenEndpoints iiop://<ipaddress>:port
2) In second terminal window: source setenv.sh and enter ./publisher -DCPSInfoRepo <ipaddress>:port
3) In third terminal window: source setenv.sh and enter ./subscriber -DCPSInfoRepo <ipaddress>:port
Movie critic messages are transmitted from publisher to subscriber window as expected

However, when I attempt to run with 2 RHEL 8.3 VMs, with the publisher and DCPSInfoRepo server running on one VM and the subscriber running on the second VM, the subscriber and publisher applications seems able to connect to the DCPSInfoRepo server but the publisher application eventually times out waiting for the subscriber and both applications end with no messages transmitted.

Here are the steps:
1) On VM1: in 1 terminal window source setenv.h and start server with DCPSInfoRepo -ORBListenEndpoints iiop://<vm1_ipaddress>:port 
2) On VM1: open second terminal window, source setenv.sh and launch publisher app with ./publisher -DCPSInfoRepo <vm1_ipaddress>:port (No error)
3) On VM2: open 1 terminal window, source setenv.sh and start subscriber app with ./subscriber -DCPSInfoRepo <vm1_ipaddress>:port (No error)
Eventually publisher times out waiting for subscriber

Also, I am able to ping between VM1 and VM2 and firewalld is disabled on both VMs.

I have combed through the Developer Guide with no success for this particular scenario.  Any assistance or recommendations from anyone who's been successful with this type of scenario is greatly appreciated.

**Comment by jrw972** [comment]
Do the logs show anything interesting like DNS failures or picking wrong network interfaces or address?
